File: Swift/ExampleTests/common/OrderItemTyped.swift
public struct OrderItemTyped: Equatable, CustomStringConvertible {
    public let name: String
    public let quantity: Int
    public let price: String
    public let itemTotal: String

    public init(name: String, quantity: Int, price: String, itemTotal: String) {
        self.name = name
        self.quantity = quantity
        self.price = price
        self.itemTotal = itemTotal
    }

    public init(from s: OrderItemString) {
        self.name = s.name
        self.quantity = Int(s.quantity) ?? 0
        self.price = s.price
        self.itemTotal = s.itemTotal
    }

    public func toStringStruct() -> OrderItemString {
        return OrderItemString(
            name: String(describing: name),
            quantity: String(describing: quantity),
            price: String(describing: price),
            itemTotal: String(describing: itemTotal)
        )
    }

    public static func toStringList(_ list: [OrderItemTyped]) -> [OrderItemString] {
        return list.map { $0.toStringStruct() }
    }

    public static func fromStringList(_ list: [OrderItemString]) -> [OrderItemTyped] {
        return list.map { OrderItemTyped(from: $0) }
    }

    public func toJSONValue() -> [String: Any] {
        return [
            "name": name,
            "quantity": quantity,
            "price": price,
            "itemTotal": itemTotal,
        ]
    }

    public func toJSON() throws -> String {
        return try Json.write(toJSONValue())
    }

    public init(fromJSONValue m: [String: Any]) throws {
        self.name = try Json.asString(Json.require(m, "name"), "name")
        self.quantity = try Json.asInt(Json.require(m, "quantity"), "quantity")
        self.price = try Json.asString(Json.require(m, "price"), "price")
        self.itemTotal = try Json.asString(Json.require(m, "itemTotal"), "itemTotal")
    }

    public init(fromJSON text: String) throws {
        try self.init(fromJSONValue: Json.parseObject(text))
    }

    public static func toJSONList(_ list: [OrderItemTyped]) throws -> String {
        return try Json.write(list.map { $0.toJSONValue() })
    }

    public static func fromJSONList(_ text: String) throws -> [OrderItemTyped] {
        return try Json.parseArray(text).map {
            try OrderItemTyped(fromJSONValue: Json.asObject($0, "OrderItemTyped"))
        }
    }

    public var description: String {
        return "Name=\(name), Quantity=\(quantity), Price=\(price), ItemTotal=\(itemTotal)"
    }
}

File: Rust/ExampleTests/common/orderitem_typed.rs
#![allow(dead_code, unused_imports, unused_variables)]

use super::json;
use super::orderitem_string::OrderItemString;

#[derive(Debug, Clone, Default, PartialEq)]
pub struct OrderItemTyped {
    pub name: String,
    pub quantity: i32,
    pub price: String,
    pub itemtotal: String,
}

impl OrderItemTyped {
    pub fn from_str_struct(s: &OrderItemString) -> Self {
        Self {
            name: s.name.clone(),
            quantity: s.quantity.parse::<i32>().unwrap_or_default(),
            price: s.price.clone(),
            itemtotal: s.itemtotal.clone(),
        }
    }

    pub fn to_str_struct(&self) -> OrderItemString {
        OrderItemString {
            name: self.name.clone(),
            quantity: self.quantity.to_string(),
            price: self.price.clone(),
            itemtotal: self.itemtotal.clone(),
        }
    }

    pub fn to_string_list(list: &[OrderItemTyped]) -> Vec<OrderItemString> {
        list.iter().map(|t| t.to_str_struct()).collect()
    }

    pub fn from_string_list(list: &[OrderItemString]) -> Vec<OrderItemTyped> {
        list.iter().map(OrderItemTyped::from_str_struct).collect()
    }

    pub fn to_json_value(&self) -> json::Value {
        json::Value::Object(vec![
            ("Name".to_string(), json::Value::Str(self.name.clone())),
            ("Quantity".to_string(), json::Value::number_from_i64(self.quantity as i64)),
            ("Price".to_string(), json::Value::Str(self.price.clone())),
            ("ItemTotal".to_string(), json::Value::Str(self.itemtotal.clone())),
        ])
    }

    pub fn to_json(&self) -> String {
        json::write(&self.to_json_value())
    }

    pub fn from_json_value(v: &json::Value) -> json::JsonResult<Self> {
        Ok(Self {
            name: json::as_string(json::require(v, "Name")?, "Name")?,
            quantity: json::as_i32(json::require(v, "Quantity")?, "Quantity")?,
            price: json::as_string(json::require(v, "Price")?, "Price")?,
            itemtotal: json::as_string(json::require(v, "ItemTotal")?, "ItemTotal")?,
        })
    }

    pub fn from_json(text: &str) -> json::JsonResult<Self> {
        Self::from_json_value(&json::parse(text)?)
    }

    pub fn to_json_list(list: &[OrderItemTyped]) -> String {
        json::write(&json::Value::Array(
            list.iter().map(|item| item.to_json_value()).collect(),
        ))
    }

    pub fn from_json_list(text: &str) -> json::JsonResult<Vec<OrderItemTyped>> {
        let root = json::parse(text)?;
        let items = json::as_array(&root, "OrderItemTyped")?;
        let mut result = Vec::with_capacity(items.len());
        for e in items {
            result.push(Self::from_json_value(e)?);
        }
        Ok(result)
    }
}
